warehouse/query-core/src/main/java/datawave/query/planner/ThreadedRangeBundlerIterator.java

```diff
@@ -30,15 +30,39 @@
 import datawave.query.tld.TLDQueryIterator;
 import datawave.query.util.count.CountMapSerDe;
 
+/**
+ * This class creates a decoupled producer/consumer of QueryData where the producer and/or consumer may be slow. The only bundling going on is the async
+ * fetching of QueryPlan from the producer
+ *
+ * The producer will begin producing immediately on construction by creating a new Thread running a RangeConsumer. The RangeConsumer will continually put to the
+ * queue, blocking if the queue is currently full (maxRanges)
+ *
+ * The consumer will be lazy initialized when hasNext() is called.
+ */
 public class ThreadedRangeBundlerIterator implements Iterator<QueryData>, Closeable {
     private static final Logger log = ThreadConfigurableLogger.getLogger(ThreadedRangeBundlerIterator.class);
 
+    /**
+     * Max time to wait to pull an item off the queue when processing hasNext() before checking rangeConsumer state. This will be the minimum time waited even
+     * for an empty iterator
```

Comment:
For example, when nothing is found in the shard index, or when the last element is found an the wrapping iterator calls hasNext which returns false.